### PR TITLE
Remove more usages of specialized collections in favor of collection exprs

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs [hunk 1]
```diff
@@ -2456,7 +2456,7 @@ private static SyntaxNode WithEqualsValue(SyntaxNode declaration, EqualsValueCla
         return declaration;
     }
 
-    private static readonly IReadOnlyList<SyntaxNode> s_EmptyList = SpecializedCollections.EmptyReadOnlyList<SyntaxNode>();
+    private static readonly IReadOnlyList<SyntaxNode> s_EmptyList = [];
```
Comment (ToddGrun) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:2459]: >private static readonly IReadOnlyList<SyntaxNode> s_EmptyList = []; [](http://example.com/codeflow?start=4&length=67)

debatable if this is worth having

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:2459]: fair :D

File: src/Compilers/Core/AnalyzerDriver/DeclarationComputer.cs [hunk 2]
```diff
@@ -31,7 +31,7 @@ internal static DeclarationInfo GetDeclarationInfo(SemanticModel model, SyntaxNo
 
         internal static DeclarationInfo GetDeclarationInfo(SemanticModel model, SyntaxNode node, bool getSymbol, SyntaxNode executableCodeBlock, CancellationToken cancellationToken)
         {
-            return GetDeclarationInfo(model, node, getSymbol, SpecializedCollections.SingletonEnumerable(executableCodeBlock), cancellationToken);
+            return GetDeclarationInfo(model, node, getSymbol, [executableCodeBlock], cancellationToken);
```
Comment (AlekseyTs) [inline src/Compilers/Core/AnalyzerDriver/DeclarationComputer.cs:34]: We generally do not welcome changes like this under Compilers.

File: src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchStatementHelpers.cs [hunk 3]
```diff
@@ -71,7 +71,7 @@ public static ICollection<ISymbol> GetMissingEnumMembers(ISwitchOperation switch
             if (!TryGetAllEnumMembers(switchExpressionType, enumMembers) ||
                 !TryRemoveExistingEnumMembers(switchStatement, enumMembers))
             {
-                return SpecializedCollections.EmptyCollection<ISymbol>();
+                return [];
```
Comment (sharwell) [inline src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchStatementHelpers.cs:74]: ❓ What type gets constructed for this case? I would expect this change introduced a new allocation.

File: src/VisualStudio/Core/Def/GenerateType/GenerateTypeDialogViewModel.cs [hunk 4]
```diff
@@ -781,7 +781,7 @@ private static string UpdateExtension(string currentFileName, string desiredFile
         }
         else
         {
-            this.ProjectFolders = SpecializedCollections.EmptyList<string>();
+            this.ProjectFolders = [];
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/GenerateType/GenerateTypeDialogViewModel.cs:784]: ⚠️ This appears to be a new allocation

File: src/VisualStudio/Core/Def/NavigationBar/NavigationBarClient.cs [hunk 5]
```diff
@@ -52,8 +52,8 @@ internal class NavigationBarClient :
 
         _workspace = workspace;
         _imageService = (IVsImageService2)serviceProvider.GetService(typeof(SVsImageService));
-        _projectItems = SpecializedCollections.EmptyList<NavigationBarProjectItem>();
-        _currentTypeItems = SpecializedCollections.EmptyList<NavigationBarItem>();
+        _projectItems = [];
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/NavigationBar/NavigationBarClient.cs:55]: ⚠️ This appears to be a new allocation

File: src/VisualStudio/Core/Def/NavigationBar/NavigationBarClient.cs [hunk 6]
```diff
@@ -52,8 +52,8 @@ internal class NavigationBarClient :
...
+        _currentTypeItems = [];
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/NavigationBar/NavigationBarClient.cs:56]: ⚠️ This appears to be a new allocation

File: src/VisualStudio/Core/Def/Venus/ContainedLanguage.IVsContainedCode.cs [hunk 7]
```diff
@@ -52,9 +52,7 @@ private IList<TextSpanAndCookie> EnumOriginalCodeBlocksWorker(CancellationToken
         var snapshot = this.SubjectBuffer.CurrentSnapshot;
         var document = snapshot.GetOpenDocumentInCurrentContextWithChanges();
         if (document == null)
-        {
-            return SpecializedCollections.EmptyList<TextSpanAndCookie>();
-        }
+            return [];
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Venus/ContainedLanguage.IVsContainedCode.cs:55]: ⚠️ This appears to be a new allocation

File: src/VisualStudio/Core/Def/Watson/FaultReporter.cs [hunk 8]
```diff
@@ -291,7 +291,7 @@ private static IList<string> CollectServiceHubLogFilePaths()
             // ignore failures
         }
 
-        return SpecializedCollections.EmptyList<string>();
+        return [];
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Watson/FaultReporter.cs:294]: ⚠️ This appears to be a new allocation

File: src/VisualStudio/Core/Def/Watson/FaultReporter.cs [hunk 9]
```diff
@@ -269,7 +269,7 @@ private static IList<string> CollectLogHubFilePaths()
...
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Watson/FaultReporter.cs:272]: ⚠️ This appears to be a new allocation

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Indentation/CSharpSmartTokenFormatter.cs [hunk 10]
```diff
@@ -84,7 +84,7 @@ public IList<TextChange> FormatToken(SyntaxToken token, CancellationToken cancel
         if (previousToken.Kind() == SyntaxKind.None)
         {
             // no previous token. nothing to format
-            return SpecializedCollections.EmptyList<TextChange>();
+            return [];
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Indentation/CSharpSmartTokenFormatter.cs:87]: ⚠️ This appears to be a new allocation

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AbstractSyntaxFormatting.cs [hunk 11]
```diff
@@ -52,7 +50,7 @@ public IFormattingResult GetFormattingResult(SyntaxNode node, IEnumerable<TextSp
 
         if (spansToFormat.Count == 0)
         {
-            return CreateAggregatedFormattingResult(node, SpecializedCollections.EmptyList<AbstractFormattingResult>());
+            return CreateAggregatedFormattingResult(node, results: []);
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AbstractSyntaxFormatting.cs:53]: ⚠️ This appears to be a new allocation

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AbstractSyntaxFormatting.cs [hunk 12]
```diff
@@ -72,7 +70,7 @@ public IFormattingResult GetFormattingResult(SyntaxNode node, IEnumerable<TextSp
         // quick simple case check
         if (results == null)
         {
-            return CreateAggregatedFormattingResult(node, SpecializedCollections.EmptyList<AbstractFormattingResult>());
+            return CreateAggregatedFormattingResult(node, results: []);
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AbstractSyntaxFormatting.cs:73]: ⚠️ This appears to be a new allocation

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/Symbols/CodeGenerationNamespaceInfo.cs [hunk 13]
```diff
@@ -38,8 +38,6 @@ public static IList<ISymbol> GetImports(INamespaceSymbol @namespace)
 
     private static IList<ISymbol> GetImports(CodeGenerationNamespaceInfo info)
     {
-        return info == null
-            ? SpecializedCollections.EmptyList<ISymbol>()
-            : info._imports;
+        return info == null ? [] : info._imports;
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/Symbols/CodeGenerationNamespaceInfo.cs:41]: ⚠️ This appears to be a new allocation

**Pull request merged**